datafusion/optimizer/src/simplify_expressions.rs

```diff
@@ -1951,7 +1951,7 @@ mod tests {
         let date_plus_interval_expr = to_timestamp_expr(ts_string)
             .cast_to(&DataType::Date32, schema)
             .unwrap()
-            + Expr::Literal(ScalarValue::IntervalDayTime(Some(123)));
```

Comment:
Based upon tests that used the parser to load "real" dates seemed to confirm the lower-order bytes are the `ms` part, so this was adding `123ms` which wasn't enough to bump the 32 bit date forward. Since the parser based tests are more likely to be correct than this hand-rolled one, I updated this to match that.